presto-hive/src/main/java/com/facebook/presto/hive/s3/PrestoS3FileSystem.java

```diff
@@ -442,8 +381,47 @@ public FSDataOutputStream create(Path path, FsPermission permission, boolean ove
         File tempFile = createTempFile(stagingDirectory.toPath(), "presto-s3-", ".tmp").toFile();
 
         String key = keyFromPath(qualifiedPath(path));
+
+        Region region = Region.US_EAST_1;
+        if (pinS3ClientToCurrentRegion) {
+            try {
+                Region currentRegion = new DefaultAwsRegionProviderChain().getRegion();
+                if (currentRegion != null) {
+                    region = currentRegion;
+                }
+            }
+            catch (Exception ignored) { }
+        }
+
+        Configuration conf = getConf();
+        String endpoint = conf.get(S3_ENDPOINT);
+
+        boolean disableChecksums = false;
+        if (endpoint != null) {
+            try {
+                URI endpointUri = URI.create(endpoint);
+                if (endpointUri.getScheme() == null) {
+                    boolean sslEnabled = conf.getBoolean(S3_SSL_ENABLED, true);
+                    endpoint = (sslEnabled ? "https://" : "http://") + endpoint;
+                    endpointUri = URI.create(endpoint);
+                }
+                disableChecksums = "http".equalsIgnoreCase(endpointUri.getScheme());
+                if (disableChecksums) {
+                    log.debug("HTTP endpoint detected for uploads: %s - disabling checksum validation", endpoint);
+                }
+            }
+            catch (IllegalArgumentException e) {
+                log.error("Invalid S3 endpoint URL: %s", endpoint);
+            }
+        }
```

Comment:
Why do we need this again? We should be handling this during S3 client creation, right?